docs/docs/guides/asynchronous-state-sync.md

```diff
@@ -2,3 +2,185 @@
 title: Asynchronous State Sync
 sidebar_label: Asynchronous State Sync
 ---
+
+Recoil [atoms](/docs/api-reference/core/atom) represent local application state.  Your application may have remote or server-side state as well, such as via a RESTful API.  Consider synchronizing the remote state with Recoil atoms.  Doing this allows you to easily access or write to the  state from React components using the `useRecoilState()` hook, or use that state as input to the Recoil data-flow graph for other derived state selectors.  If you're looking to [query a database or server for read-only data](asynchronous-data-queries), consider using asynchronous selectors.
+
+## Local State Example
+
+This example provides the friend status as local state only.
+
+```jsx
+const currentUserIDState = atom({
+  key: 'CurrentUserID',
+  default: null,
+});
+
+function CurrentUserInfo() {
+  const [currentUserID] = useRecoilState(currentUserIDState);
+  return <div>Current User: {currentUserID}</div>;
+}
+```
+
+## Sync State From Server
+
+We can subscribe to asynchronous changes in the remote state and update the atom value to match.  This can be done using standard React `useEffect()` hook or other popular libraries.
+
+```jsx
+function CurrentUserIDSubscription() {
+  const setCurrentUserID = useSetRecoilState(currentUserIDState);
+
+  useEffect(() => {
+    RemoteStateAPI.subscribeToCurrentUserID(setCurrentUserID);
+    // Specify how to cleanup after this effect
+    return function cleanup() {
+      RemoteServerAPI.unsubscribeFromFriendStatus(setCurrentUserID);
+    };
+  }, []);
+
+  return null;
+}
+
+function MyApp() {
+  return (
+    <RecoilRoot>
+      <CurrentUserIDSubscription />
+      <CurrentUserInfo />
+    </RecoilRoot>
+  );
+}
+```
+
+If you want to handle synchronization of multiple atoms in a single place, you can also use the [State Persistence](persistence) pattern.
+
+## Bi-Directional Synching
+
+You can also sync the state so local changes are updated on the server:
+
+```jsx
+function CurrentUserIDSubscription() {
+  const [currentUserID, setCurrentUserID] = useRecoilState(currentUserIDState);
+  let knownServerCurrentUserID;
+
+  // Subscribe server changes to update atom state
+  useEffect(() => {
+    function handleUserChange(id) {
+      knownServerCurrentUserID = id;
+      setCurrentUserID(id);
+    }
+
+    RemoteStateAPI.subscribeToCurrentUserID(handleUserChange);
+    // Specify how to cleanup after this effect
+    return function cleanup() {
+      RemoteServerAPI.unsubscribeFromFriendStatus(handleUserChange);
+    };
+  }, []);
+
+  // Subscribe atom changes to update server state
```

Comment:
I wonder if it's worth making this a little bit more robust, e.g. by including timestamp comparisons, or version counters.
Right now I think there is a race condition if the Recoil state is updated at roughly the same time that the remote values changes.

Reply:
It also suffers from redundant setting of the local atom if the user sets it and the server mirrors back the change to all subscribers, though the value should be correct.  Yeah, just wanted a simplified example, but I'm not very keen on it so far.  Could definitely be improved.